packages/jira-adapter/src/filters/dashboard/gadget.ts

```
@@ -0,0 +1,151 @@
+  const dashboardId = getParents(instance)[0].value.value.id
+
+  const response = await client.getSinglePage({
+    url: `/rest/api/3/dashboard/${dashboardId}/items/${instance.value.id}/properties/${key}`,
+  })
+  if (Array.isArray(response.data)) {
+    log.error(`Invalid response from server when fetching property '${key}' value for ${instance.elemID.getFullName()}: ${safeJsonStringify(response.data)}`)
+    return undefined
+  }
+
+  return response.data.value
+}
+
+const filter: FilterCreator = ({ client, config }) => ({
+  onFetch: async elements => {
+    await Promise.all(elements
+      .filter(isInstanceElement)
+      .filter(instance => instance.elemID.typeName === DASHBOARD_GADGET_TYPE)
+      .map(async instance => {
+        const keys = await getPropertiesKeys(instance, client)
+        instance.value.properties = _.pickBy(Object.fromEntries(
+          await Promise.all(keys.map(async key => [
+            key,
+            await getPropertyValue(instance, key, client),
+          ]))
+        ), values.isDefined)
+      }))
+
+    const gadgetType = findObject(elements, DASHBOARD_GADGET_TYPE)
+    if (gadgetType === undefined) {
+      log.warn(`${DASHBOARD_GADGET_TYPE} type not found`)
+      return
+    }
+    setFieldDeploymentAnnotations(gadgetType, 'properties')
+  },
+
+  deploy: async changes => {
+    const [relevantChanges, leftoverChanges] = _.partition(
+      changes,
+      change => isInstanceChange(change)
+        && getChangeData(change).elemID.typeName === DASHBOARD_GADGET_TYPE
+    )
+
+
+    const deployResult = await deployChanges(
+      relevantChanges
+        .filter(isInstanceChange),
+      async change => {
+        try {
+          await defaultDeployChange({
+            change,
+            client,
+            apiDefinitions: config.apiDefinitions,
+            fieldsToIgnore: [
+              'properties',
+            ],
+          })
+        } catch (err) {
+          if (err instanceof clientUtils.HTTPError
```

And if its 500, don't you want to rethrow the error?